bundles/framework/language-selector/components/LanguageChanger.jsx

```diff
@@ -1,18 +1,18 @@
 import React from 'react';
-import {LanguageSelect} from './LanguageSelect';
-import {getSupportedLanguages} from './getSupportedLanguages';
+import { LanguageSelect } from './LanguageSelect';
+import { getSupportedLanguages } from './getSupportedLanguages';
 
-const handleLanguageChange = (language) => {
+const handleLanguageChange = (event) => {
     let redirect;
-    const newLang = 'lang=' + language;
+    const newLang = 'lang=' + event.target.value;
     if (window.location.href.indexOf('?') === -1) {
         redirect = window.location.href + '?' + newLang;
     } else if (window.location.href.indexOf('lang=') > -1) {
         redirect = window.location.href.replace(/lang=[^&]+/, newLang);
     } else {
         redirect = window.location.href + '&' + newLang;
     }
-    window.location.href = redirect;
+    window.location.href = redirect.replace('#', '');
```

Comment:
Should we re-construct the URL from location.path + modified location.search + location.hash instead?